pallets/amm/src/lib.rs

```diff
@@ -610,10 +613,46 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
             .ok_or(ArithmeticError::Overflow)?)
     }
 
+    #[allow(dead_code)]
+    fn update_oracle(
+        pool: &mut Pool<AssetIdOf<T, I>, BalanceOf<T, I>, T::BlockNumber>,
+    ) -> Result<(), DispatchError> {
+        let block_timestamp = frame_system::Pallet::<T>::block_number();
+
+        if pool.block_timestamp_last != block_timestamp {
+            pool.block_timestamp_last = block_timestamp;
+            let time_elapsed: T::BlockNumber =
+                block_timestamp.saturating_sub(pool.block_timestamp_last);
+
+            let price0_fraction: BalanceOf<T, I> =
+                FixedU128::saturating_from_rational(pool.quote_amount, pool.base_amount)
+                    .into_inner();
+            let price1_fraction: BalanceOf<T, I> =
+                FixedU128::saturating_from_rational(pool.base_amount, pool.quote_amount)
+                    .into_inner();
+
+            pool.price_0_cumulative_last =
+                pool.price_0_cumulative_last.saturating_add(price0_fraction);
```

Comment:
not good to have so many saturating I think, can we find another way to make it not overflow?

Reply:
let me try @GopherJ